### DISCUSSION TITLE
request to enable mirror of repository

### DISCUSSION THREAD

**Discussion by dghgit** [community/community#40494] [Repositories]
### Select Topic Area

General

### Body

The arrival of post-quantum algorithms has led to an explosion in the size of our test files. In an attempt (possibly futile..) to try and keep things under control we  have set up bc-test-data as a repository and we would like to have it mirror

https://www.bouncycastle.org/repositories/bc-test-data

In the same way our other ones do. If there are any questions, please let me know.

Thanks,

David

**Comment by ossanna16** [comment]
Hi there @dghgit and welcome to our community! Thank you for the suggestion!

**Reply by dghgit** [reply]
Hi Anna ( @ossanna16 ), thanks for noticing my message and apologies for any confusion caused, but this is a request not a suggestion. I probably should have been clearer but previously I have emailed the support address, I guess in this case there is not enough context.

If you look at https://github.com/bcgit you will see that the "bc-" repositories are all mirrors. This is largely because, as they're cryptography APIs, we try to minimize the chance of something being published by accident by introducing a middle step before publishing to github from our primary server in Australia and in the early days after discussion with team at github it was agreed this was the easiest way to deal with this issue.

Usually what I would do is set up an empty repository, alert support, and then they would perform the necessary magic and convert the empty repository to a mirror based on the URL I provided.

In this case the empty repository is at:

https://github.com/bcgit/bc-test-data

and the URL for the mirror repository is:

https://www.bouncycastle.org/repositories/bc-test-data

Would you bring this to the appropriate groups attention and ask them to set it up? As I mentioned in my original message test data around post quantum algorithms is huge, there's another round of algorithm development going now, NIST have announced which will start making itself felt in March next year and we expect other organisations to start jumping on the bandwagon soon to, so we're trying to find a better way of managing the files, which are used across our APIs, before all of us get crushed under the size of them. It would be greatly appreciated if GitHub would be able to assist us in this matter by setting up the new mirror repository.

If you need any further details, please let me know. Appreciate your assistance.

Thanks,

David

**Reply by ossanna16** [reply]
Thank you for clarifying @dghgit!
